packages/frontend/src/app/search/interpretation/interpretation.service.spec.ts

```diff
@@ -0,0 +1,146 @@
+          inputs.push(query + service.inputSeparator + filter.name + value); // no filter separator
+        } else
+          inputs +=
+            service.inputSeparator +
+            filter.name +
+            service.filterSeparator +
+            value;
+      });
+    });
+    return inputs;
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(InterpretationService);
+    service.filters = possiblefilters;
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should parse and return the right filters', () => {
+    const inputs: string = getPossibleInputs(false) as string;
+    const interpretedInput = service.parse(inputs);
+    const splittedInputs = inputs.split(service.inputSeparator);
+    const query = splittedInputs.shift();
+
+    expect(query).toEqual(interpretedInput.query);
+
+    splittedInputs.forEach((input, index) => {
+      const filter = interpretedInput.filters[index];
+      expect(filter).not.toBeUndefined();
+
+      const splittedInput = input.split(service.filterSeparator);
+      expect(filter.name).toEqual(splittedInput[0]);
+      expect(filter.value).toEqual(splittedInput[1]);
+    });
+  });
+
+  it('should parse a wrongly positioned query', () => {
+    const query = 'query';
+    const filter1 =
+      possiblefilters[0].name +
+      service.filterSeparator +
+      possiblefilters[0].possibleValues[0];
+    const filter2 =
+      possiblefilters[0].name +
+      service.filterSeparator +
+      possiblefilters[0].possibleValues[1];
+
+    // query between filters
+    const input: string =
+      filter1 +
+      service.inputSeparator +
+      query +
+      service.inputSeparator +
+      filter2;
+
+    const interpretedInput = service.parse(input);
+
+    expect(interpretedInput.query).toEqual(query);
+    expect(interpretedInput.filters.length).toEqual(2);
+
+    interpretedInput.filters.forEach((filter, index) => {
+      expect(filter.name).toEqual(possiblefilters[0].name);
+      expect(filter.value).toEqual(possiblefilters[0].possibleValues[index]);
+    });
+  });
+
+  it('should not be case-sensitive', () => {
+    const inputs: string = (getPossibleInputs(false) as string).toUpperCase(); // upper case input
+    const interpretedInput = service.parse(inputs);
+
+    const splittedInputs = inputs.split(service.inputSeparator);
+    const query = splittedInputs.shift();
+
+    expect(query).toEqual(interpretedInput.query);
+
+    splittedInputs.forEach((input, index) => {
+      const filter = interpretedInput.filters[index];
+      expect(filter).not.toBeUndefined();
+
+      // the filters should have been lowercased
+      const splittedInput = input.toLowerCase().split(service.filterSeparator);
+      expect(filter.name).toEqual(splittedInput[0]);
+      expect(filter.value).toEqual(splittedInput[1]);
+    });
+  });
+
+  it('should not parse wrong filters', () => {
```

Comment:
thank you for continuing to write good tests, I can't underscore how much this will be important to the future health of this project 😄